components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -85,6 +85,9 @@ private void validate(String username, String userStoreDomain, String tenantDoma
 
         String userId = UserSessionStore.getInstance().getUserId(username, getTenantId(tenantDomain),
```

Comment:
What happens when we do this call?
UserSessionStore.getInstance().getUserId(username, getTenantId(tenantDomain), userStoreDomain);

Reply:
@pamodaaw can we replace this with the below logic? will this method be called for federated users by anychance?

Reply:
> What happens when we do this call?
> UserSessionStore.getInstance().getUserId(username, getTenantId(tenantDomain), userStoreDomain);

That getUserID method call is to return the user Id of a user from the database table of IDN_AUTH_USER. Therefore it returns null.

Reply:
Yes, we can remove the user-id validation from the IDN_AUTH_USER table since we add only the federated user IDs to this table.